test/external/testcorenrn/CMakeLists.txt

```diff
@@ -105,6 +105,45 @@ foreach(test ${spike_comparison_tests})
     endif()
     # Set up a test that runs the simulation using CoreNEURON's offline mode. First we must run
     # NEURON and tell it to dump the model without simulating it.
+    if("${test}" STREQUAL "patstim")
+      nrn_add_test(
+        GROUP testcorenrn_${test}
+        NAME coreneuron_${processor}_offline_saverestore
+        REQUIRES coreneuron mpi ${processor}
+        CONFLICTS mpi_dynamic
+        PROCESSORS ${local_mpi_ranks}
+        COMMAND bash coreneuron_${processor}_offline_saverestore.sh
+        OUTPUT asciispikes::out.dat)
+      if((${CORENRN_ENABLE_GPU} STREQUAL "ON" AND "${processor}" STREQUAL "gpu")
+         OR NOT "${processor}" STREQUAL "gpu")
```

Comment:
Not sure I understand the logic here? It looks like this reduces to "always on CPU, on GPU if it's enabled" which is what you already have from `REQUIRES ... ${processor}` just above 🤔

Reply:
Aha, I understand now 👍 

I would say it's fine to write a few extra bash scripts for tests that are not enabled.